analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -301,7 +301,10 @@ open class Npm(
 
         log.info { "Searching for 'package.json' files in '$nodeModulesDir'..." }
 
-        val nodeModulesFiles = nodeModulesDir.walk().filter {
+        // PNPM symlinks workspace modules in the `node_modules` directory, which will result in cyclic symlinks of
+        // `node_module` directories. Limit the search depth to '2' in this case.
+        val searchDepth = if (managerName == "PNPM") 2 else Int.MAX_VALUE
```

Comment:
I am not sure how much effort it means, but it may be possible to avoid these special treatments for PNPM by designing the `Npm` base class accordingly for being extended. For instance, `Npm` could define a function/property `modulesSearchDepth`, which could be overridden by `Pnpm`.